src/main/java/software/amazon/cloudformation/LambdaWrapper.java

```diff
@@ -252,9 +254,16 @@ public void handleRequest(final InputStream inputStream, final OutputStream outp
             String input = this.serializer.decompress(IOUtils.toString(inputStream, StandardCharsets.UTF_8));
 
             JSONObject rawInput = new JSONObject(new JSONTokener(input));
+            bearerToken = rawInput.getString("bearerToken");
 
             // deserialize incoming payload to modelled request
-            request = this.serializer.deserialize(input, typeReference);
+            try {
+                request = this.serializer.deserialize(input, typeReference);
+            } catch (MismatchedInputException e) {
+                JSONObject resourceSchemaJSONObject = provideResourceSchemaJSONObject();
+                JSONObject rawModelObject = rawInput.getJSONObject("requestData").getJSONObject("resourceProperties");
```

Comment:
What happens to exceptions thrown here, if the JSON payload is completely malformed and doesn't contain the fields you're expecting?

Reply:
1. The JSON payload completely malformed only if it invoked not by CloudFormation, which this case is really unlikely.

2. The above `this.serializer.deserialize(input, typeReference)` will throw different exception if the payload is malformed, it won't be catched by `MismatchedInputException`. We have unit test to cover the malformed input (0 bytes or invalid json),  those passed as expected.

3.  The only case I can think that `rawInput.getJSONObject("requestData").getJSONObject("resourceProperties");` is `null` or malformed `requestData`, the whole payload `HandlerRequest` has `MismatchedInputException`  in other fields.  However, I don't think it's case we need to worry about.